core/trino-main/src/test/java/io/trino/operator/aggregation/TestAggregationMaskCompiler.java

```diff
@@ -197,6 +204,11 @@ private static Block createMaskBlockRle(int positionCount, byte mask)
         return RunLengthEncodedBlock.create(createMaskBlock(1, new byte[] {mask}), positionCount);
     }
 
+    private static Block createMaskBlockAsDictionary(int positionCount, byte[] mask)
+    {
+        return DictionaryBlock.create(positionCount, new ByteArrayBlock(positionCount, Optional.empty(), mask), IntStream.range(0, positionCount).toArray());
```

Comment:
The ids here match the positions, so this may not catch the position problems you fixed above.  Maybe do something like:
```
int[] ids = IntStream.concat(IntStream.concat(IntStream.of(100, 101, 102), IntStream.range(positionCount, 0)), IntStream.of(200, 201, 202)).toArray();
```
then  do `getRegion(3, positionCount)`